muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/transcriptofrecords/rest/TranscriptofRecordsRESTService.java

```diff
@@ -414,6 +468,46 @@ public Response retrieveForm(@PathParam("USERIDENTIFIER") String userIdentifierS
     return Response.ok(response).build();
   }
 
+  @POST
+  @Path("/plannedCourses/")
+  @RESTPermit(handling=Handling.INLINE)
+  public Response planCourse(
+      VopsPlannedCourseRESTModel model
+  ) {
+    // TODO check for permission
```

Comment:
TODO

If users can only plan for themselves (f.ex. guider cannot modify) then just using logged user instead of model-provided student identifier is enough for a permission check. Otherwise check for guider permission or the model.studentidentifier == loggeduser.